x-pack/plugins/infra/server/lib/adapters/framework/kibana_framework_adapter.ts

```diff
@@ -149,8 +149,17 @@ export class InfraKibanaBackendFrameworkAdapter implements InfraBackendFramework
   ) {
     const internalRequest = req[internalInfraFrameworkRequest];
     const server = internalRequest.server;
+
+    let url = '/api/metrics/vis/data';
+    if (server.plugins.spaces) {
+      const spaceId = server.plugins.spaces.getSpaceId(internalRequest);
+      if (spaceId !== 'default') {
+        url = `/s/${spaceId}${url}`;
```

Comment:
Sorry for the drive-by review, I saw it come across in slack: I don't fully understand this code, but can you use `internalRequest.getBasePath()` instead of deriving a space-aware URL yourself?

If that works, this should become something like 
```js
let url = `${internalRequest.getBasePath()}/api/metrics/vis/data`;
```

and then you can remove the check for the spaces plugin altogether.


Alternatively, if it's possible to make this request client-side, then you can get space-aware URLs for free there too. This would have the benefit of "just working" when you migrate to the New Platform too, as I don't expect they'll want to expose `server.inject` to plugins anymore.

Reply:
These are very valid comments, thank you.

The issue here is that `server.inject` seems to inject the request below any base path that might be configured, so adding the full base path (random thing + space, if applicable) does not work.

The request is server-side because we're hiding the functionality behind a GraphQL endpoint.

I agree this is something we need to tackle for the new platform, but that's beyond the scope of this bugfix.

@simianhacker what do you think?

Reply:
Thanks for clarifying @skh, I hadn't considered the server base path being stripped out of the `server.inject` handling.

I totally understand that refactoring is out of scope for this PR 👍

Reply:
I assume after the Platform refactor/upgrade we will need to re-evaluate how the internal requests work. I'm sure the platform team will provide guidance on how they expect this to work.